File: 프로그래머스/1/176963. 추억 점수/추억 점수.java
import java.util.*;

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        Map<String, Integer> yearnMap = new HashMap<>();
        for (int i = 0; i < name.length; i++) {
            yearnMap.put(name[i], yearning[i]);
        }
        
        return Arrays.stream(photo)
            .mapToInt(arr ->
                Arrays.stream(arr)
                    .mapToInt(person -> yearnMap.getOrDefault(person, 0))
                    .sum()
            ).toArray();
    }
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.kt
class Solution {
    fun solution(name: Array<String>, yearning: IntArray, photo: Array<Array<String>>): IntArray {
        val yearnMap = name.zip(yearning.toTypedArray()).toMap()
        
        return photo.map { people ->
            people.sumOf { person -> yearnMap[person] ?: 0 }
        }.toIntArray()
    }
}